subsys/bluetooth/mesh/shell/shell_utils.c

```diff
@@ -37,24 +37,10 @@ static void model_instances_get(uint16_t id, struct shell_model_instance *arr, u
 	}
 }
 
-static uint8_t str2u8(const char *str)
-{
-	if (isdigit((unsigned char)str[0])) {
-		return strtoul(str, NULL, 0);
-	}
-
-	return (!strcmp(str, "on") || !strcmp(str, "enable"));
-}
-
-bool shell_model_str2bool(const char *str)
-{
-	return str2u8(str);
-}
-
-
-int shell_model_str2sensorval(const char *str, struct sensor_value *out)
+struct sensor_value shell_model_str2sensorval(const char *str, int *err)
```

Comment:
I see you did this to align with all the others, but I'd really like for it to be the other way around. It's pretty standard for C APIs to return the error code and pass the value back in a pointer, not the other way around. I do realize it is a bit too late to say this now, so I guess it's fine as it is.

Reply:
Yes the initial plan was to return the error and pass value by ref. This would have made the merging of error checking easier to. The reason why I return the value (and probably why strtol and strtul does the same) is that it is problematic to convert between types when passing by ref. If you pass a uint8_t value reference and the internal conversion type is long it will violate memory boundaries when it writes to the uint8_t memory space. This would not be the case for this particular fonction, but as you mentioned it was done to align it with the other functions.